src/connectors/toggleRefinement/connectToggleRefinement.js

```diff
@@ -277,23 +274,28 @@ export default function connectToggleRefinement(renderFn, unmountFn = noop) {
       },
 
       getWidgetSearchParameters(searchParameters, { uiState }) {
+        const state = searchParameters
+          .setQueryParameters({
+            disjunctiveFacetsRefinements: {
+              ...searchParameters.disjunctiveFacetsRefinements,
+              [attribute]: [],
+            },
+          })
+          .addDisjunctiveFacet(attribute)
+          .removeDisjunctiveFacetRefinement(attribute, off)
+          .removeDisjunctiveFacetRefinement(attribute, on);
```

Comment:
The `remove` calls are redundant with the `setQueryParameters` call. We can remove one or the other but we have to preserve the empty array when the `uiState` is empty. Here is how I would have done it:

```js
const widget = {
  getWidgetSearchParameters(searchParameters, { uiState }) {
    const withFacetConfiguration = searchParameters
      .clearRefinements(attribute)
      .addDisjunctiveFacet(attribute);

    const isRefined = Boolean(uiState.toggle && uiState.toggle[attribute]);

    // case where it's refined
    if (isRefined) {
      return withFacetConfiguration.addDisjunctiveFacetRefinement(
        attribute,
        on
      );
    }

    // case where it's not refined but with an off value
    if (hasAnOffValue) {
      return withFacetConfiguration.addDisjunctiveFacetRefinement(
        attribute,
        off
      );
    }

    // case where it's not refined withtout an off value
    return withFacetConfiguration.setQueryParameters({
      disjunctiveFacetsRefinements: {
        ...searchParameters.disjunctiveFacetsRefinements,
        [attribute]: []
      }
    });
  }
};
```